src/celeritas/optical/PreGenData.hh

```diff
@@ -39,7 +41,7 @@ struct OpticalBufferSize
  *
  * At least one of cerenkov and scintillation must be enabled.
  */
-struct OpticalGenSetup
+struct PreGenOptions
```

Comment:
Similarly for some of these other `Optical*` --> `PreGen*`, perhaps e.g. `Generator*` or `PrimaryGen*`?

Reply:
So far I've only found that `OpticalDistributionData` is used by `ScintillationGenerator` or `CerenkovGenerator`. The other classes I've marked as `PreGen` I've found to be only used on the core physics side of the code, the `Generators` only on the optical side, and `OpticalDistributionData` and `OpticalCollector` to be the interface between the two. It may be worthwhile to leave such classes in the core namespace and have only the ones operating in the optical loop be in the optical namespace?